### PR TITLE
Documentation update for react$ and react$$

### LINKED ISSUE #3943: Using react$ is returning 'method is not implemented' error
WebdriverIO version: 5.8.3

I have just started to look at the new react$ and I I'm getting an error 'Method is not implemented'. I believe this is from Appium. Firstly I wondered if this should work with React Native (we're on v0.59) and Appium? I believe it should work as there is a virtual DOM to target (I can see using react-devtools). If I look at webdriverio I see it takes a props and state object and returns an element like so...

```
        react$(
            props?: object,
            state?: object
        ): Element;
```
So with this i should be able to do (given an element exists on the app with a prop of testID = 'select.your.department')

      browser.react$({ testID: 'select.your.department' }).click()

However when i run this, I'm getting an error 'Method is not implemented'. Whether this is an error with my implementation as I can see the code is there i'm not sure.

Also looking at the documentation it seems that you need to pass in a component name, but it doesn't look to be the case with the above. Are these new commands still work in progress, or if anyone has any further info it would be greatly appreciated. 

Tried using the webdriverio gitter channel but no one had any input :(

I grabbed the appium logs but I'm not sure it tells us anything...

`0-0] 2019-05-07T23:47:18.753Z DEBUG webdriver: request failed due to response error: unknown error
[0-0] 2019-05-07T23:47:18.754Z INFO webdriver: Retrying 1/3
[0-0] 2019-05-07T23:47:18.754Z INFO webdriver: [POST] http://127.0.0.1:4723/wd/hub/session/b780e516-b1ad-4725-a5d9-fce0aca771c7/execute/sync
[0-0] 2019-05-07T23:47:18.754Z INFO webdriver: DATA { script:
    '!function(t,e){"object"==typeof exports&&"object"==typeof module?module.exports=e():"function"==typeof define&&define.amd?define([],e):"object"==typeof exports?exports.resq=e():
(t.window=t.window||{},t.window.resq=e())}(window,function(){return function(t){var e={};function n(r){if(e[r])return e[r].exports;var o=e[r]={i:r,l:!1,exports:{}};return t[r].call(o
.exports,o,o.exports,n),o.l=!0,o.exports}return n.m=t,n.c=e,n.d=function(t,e,r){n.o(t,e)||Object.defineProperty(t,e,{enumerable:!0,get:r})},n.r=function(t){"undefined"!=typeof 
Symbol&&Symbol.toStringTag&&Object.defineProperty(t,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(t,"__esModule",{value:!0})},n.t=function(t,e){if(1&e&&(t=n(t)),8&e)retu
rn t;if(4&e&&"object"==typeof t&&t&&t.__esModule)return t;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:t}),2&e&&"string"!=typeof t)for(v
ar o in t)n.d(r,o,function(e){return t[e]}.bind(null,o));return r},n.n=function(t){var e=t&&t.__esModule?function(){return t.default}:function(){return t};return n.d(e,"a",e),e},n.o=
function(t,e){return Object.prototype.hasOwnProperty.call(t,e)},n.p="",n(n.s=4)}([function(t,e){var n;n=function(){return this}();try{n=n||new Function("return this")()}catch(t){"obj
ect"==typeof window&&(n=window)}t.exports=n},function(t,e,n){"use strict";(function(t){function r(e=5e3,n){if(t.isReactLoaded)return Promise.resolve("React already loaded");return ne
w Promise((r,o)=>{let i=!1;const u=()=>{const e=(()=>{const t=document.createTreeWalker(document);if(n)return document.querySelector(n);for(;t.nextNode();)if(t.currentNode.hasOwnProp
erty("_reactRootContainer"))return t.currentNode})();if(e)return t.isReactLoaded=!0,t.rootReactElement=e._reactRootContainer._internalRoot.current,r();i||setTimeout(u,200)};u(),setTi
meout(()=>{i=!0,o("Timed out")},e)})}n.d(e,"a",function(){return r})}).call(this,n(0))},function(t,e,n){"use strict";var r=Array.isArray,o=Object.keys,i=Object.prototype.hasOwnProper
ty;t.exports=function t(e,n){if(e===n)return!0;if(e&&n&&"object"==typeof e&&"object"==typeof n){var u,c,s,f=r(e),a=r(n);if(f&&a){if((c=e.length)!=n.length)return!1;for(u=c;0!=u--;)if
(!t(e[u],n[u]))return!1;return!0}if(f!=a)return!1;var l=e instanceof Date,d=n instanceof Date;if(l!=d)return!1;if(l&&d)return e.getTime()==n.getTime();var p=e instanceof RegExp,h=n i
nstanceof RegExp;if(p!=h)return!1;if(p&&h)return e.toString()==n.toString();var y=o(e);if((c=y.length)!==o(n).length)return!1;for(u=c;0!=u--;)if(!i.call(n,y[u]))return!1;for(u=c;0!=u
--;)if(!t(e[s=y[u]],n[s]))return!1;return!0}return e!=e&&n!=n}},function(t,e,n){"use strict";var r=n(2),o=n.n(r);const{isArray:i}=Array,{keys:u}=Object;function c(t){return"function"
==typeof t}function s(t){return t instanceof HTMLElement||t instanceof Text}function f(t){return"object"==typeof t&&!i(t)}function a(t,e,n=!1){return!(!i(t)||!i(e))&&(n?t.length===e.
length&&!t.find(t=>!e.includes(t)):t.some(t=>e.includes(t)))}function l(t){let e={children:[]};if(!t)return e;var n;e.name=c(n=t.type)?n.name:n,e.props=function(t){if(!t||"string"==t
ypeof t)return t;const e={...t};return delete e.children,e}(t.memoizedProps),e.state=function(t){if(!t)return;const{baseState:e}=t;return e||t}(t.memoizedState);let{child:r}=t;if(r)f
or(e.children.push(r);r.sibling;)e.children.push(r.sibling),r=r.sibling;return e.children=e.children.map(t=>l(t)),c(t.type)&&function(t){return t.children.length>1}(e)?(e.node=functi
on(t){return t.children.map(t=>t.node).filter(t=>!!t)}(e),e.isFragment=!0):e.node=function(t){return s(t.stateNode)?t.stateNode:t.child&&s(t.child.stateNode)?t.child.stateNode:null}(
t),e}function d(t,e,n=!1,r){let o=[e];return t.forEach(t=>{o=function(t,e,n=!1){let r=[];for(;t.length||n&&!r.length;){const{children:n}=t.shift();n&&n.length&&n.forEach(n=>{e(n)&&r.
push(n),t.push(n)})}return r}(o,e=>r&&"function"==typeof r?r(e):e.name===t,n)}),o}function p(t,e,n,r=!1){return c(n)?(console.warn("Functions are not supported as filter matchers"),[
]):t.filter(t=>f(n)&&function t(e={},n={},r=!1){let 
i=[];return!u(e).length||(r?o()(e,n):(u(e).forEach(r=>{n.hasOwnProperty(r)&&(f(e[r])&&f(n[r])&&(i=i.concat(t(e[r],n[r]))),(e[r]===n[r]||a(e[r],n[r]))&&i.push(n))}),!!i.filter(t=>t).l
ength))}(n,t[e],r)||i(n)&&a(n,t[e],r)||t[e]===n)}n.d(e,"a",function(){return m});class h extends Array{constructor(t){t||(t=[]),super(...t)}byProps(t,{exact:e}={exact:!1}){const n=p(
this,"props",t,e);return new h(n)}byState(t,{exact:e}={exact:!1}){const n=p(this,"state",t,e);return new h(n)}}class y extends Object{constructor(t,e){super(t),this._nodes=e;for(let 
e in t)this[e]=t[e]}byProps(t,{exact:e}={exact:!1}){const n=p(this._nodes,"props",t,e)[0];return new y(n,this._nodes)}byState(t,{exact:e}={exact:!1}){const n=p(this._nodes,"state",t,
e)[0];return new y(n,this._nodes)}}class m{constructor(t,e){this.selectors=t.split(" ").filter(t=>!!t).map(t=>t.trim()),this.rootComponent=e,this.tree=l(this.rootComponent)}find(){re
turn this.nodes=new h(d(this.selectors,this.tree,!0)),new y(this.nodes[0],this.nodes)}findAll(){return new h(d(this.selectors,this.tree))}}},function(t,e,n){"use strict";n.r(e),funct
ion(t){n.d(e,"resq$",function(){return u}),n.d(e,"resq$$",function(){return c});var r=n(3),o=n(1);function i(e,n="find"){if(!t.isReactLoaded)throw new Error("Could not find the root 
element of your application");return new r.a(e,t.rootReactElement)[n]()}function u(t){return i(t)}function c(t){return i(t,"findAll")}n.d(e,"waitToLoadReact",function(){return o.a})}
.call(this,n(0))}])});',
   args: [] }`

### PR DESCRIPTION
**Pull request by Raulster24**
## Proposed changes

[//]: # (Added 'selector' as the parameter in the documentation of react$. Please update to a better description for this parameter, if required.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

Resolve #3943

### CODE REVIEW

**Comment by Raulster24** [comment]
@baruchvlz Please see if this is valid change.

**Comment by christian-bromann** [comment]
Can you please sign the CLA?

**Comment by Raulster24** [comment]
> Can you please sign the CLA?

Done!

**Comment by baruchvlz** [comment]
Yea, saw this was missing as well. Thanks @Raulster24.

File: packages/webdriverio/src/commands/browser/react$$.js [hunk 1]
```diff
@@ -15,7 +15,8 @@
     });
  * </example>
  *
- * @alias browser.react$
+ * @alias browser.react$$
+ * @param {String=} selector React elements or HTML elements
  * @param {Object=} props  React props the element should contain
  * @param {Object=} state  React state the element should be in
  * @return {Element}
```
Comment (CrispusDH) [inline packages/webdriverio/src/commands/browser/react$$.js]: does it return one Element, I suppose it should be `Element[]`

File: packages/webdriverio/src/commands/browser/react$$.js [hunk 2]
```diff
@@ -15,7 +15,8 @@
     });
  * </example>
  *
- * @alias browser.react$
+ * @alias browser.react$$
+ * @param {String=} selector React elements or HTML elements
```
Comment (CrispusDH) [inline packages/webdriverio/src/commands/browser/react$$.js]: should it be `String|Function` like for common selector?

Comment (christian-bromann) [reply packages/webdriverio/src/commands/browser/react$$.js]: no, this only takes a selector as string

File: packages/webdriverio/src/commands/browser/react$$.js (hunk 1 above)
Comment (Raulster24) [reply packages/webdriverio/src/commands/browser/react$$.js]: @CrispusDH  Yes, it should be `Element[]`, it returns an array of elements. I will do the changes.
@baruchvlz Please confirm.

File: packages/webdriverio/src/commands/browser/react$$.js (hunk 2 above)
Comment (christian-bromann) [inline packages/webdriverio/src/commands/browser/react$$.js]:
```suggestion
 * @param {String=} selector of React component
```

File: packages/webdriverio/src/commands/browser/react$.js [hunk 3]
```diff
@@ -20,6 +20,7 @@
  * </example>
  *
  * @alias browser.react$
+ * @param {String=} selector React elements or HTML elements
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/browser/react$.js]:
```suggestion
 * @param {String=} selector of React component
```

File: packages/webdriverio/src/commands/browser/react$$.js (hunk 2 above)
Comment (Raulster24) [reply packages/webdriverio/src/commands/browser/react$$.js]: @CrispusDH It should be `String ` only, I am not sure what you meant by `Function `here for common selectors.

File: packages/webdriverio/src/commands/browser/react$.js [hunk 4]
```diff
@@ -20,9 +20,10 @@
  * </example>
  *
  * @alias browser.react$
+ * @param {String=} selector of React component
  * @param {Object=} props  React props the element should contain
  * @param {Object=} state  React state the element should be in
- * @return {Element}
+ * @return {Element[]}
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/browser/react$.js]: not sure why this change
```suggestion
 * @return {Element}
```

**Comment by baruchvlz** [comment]
For some reason the typings display the return as `Promise<WebdriverIO.Element>` not sure if this is intended or if it's wrong. @christian-bromann

**Comment by christian-bromann** [comment]
> For some reason the typings display the return as `Promise<WebdriverIO.Element>` not sure if this is intended or if it's wrong

If you use wdio testrunner in sync mode you need to import typings as follows:
```
"types": ["node", "@wdio/sync"]
```
see https://webdriver.io/docs/typescript.html

**Pull request merged**